lib/plugins/aws/package/compile/functions.js

```diff
@@ -574,6 +578,31 @@ class AwsCompileFunctions {
 
         cfTemplate.Resources[aliasLogicalId] = aliasResource;
       }
+
+      if (functionObject.snapStart) {
+        if (!shouldVersionFunction) delete versionResource.DeletionPolicy;
+
+        const snapStartConfig = {};
+        snapStartConfig.ApplyOn = functionObject.snapStart ? 'PublishedVersions' : 'None';
+        functionResource.Properties.SnapStart = snapStartConfig;
+
+        const aliasLogicalId = this.provider.naming.getLambdaSnapStartAliasLogicalId(functionName);
+        const aliasName = this.provider.naming.getLambdaSnapStartEnabledAliasName();
+
+        functionObject.targetAlias = { name: aliasName, logicalId: aliasLogicalId };
```

Comment:
If also provisioned concurrency is setup, it'll override this setting.

I think in this case we should rely on the alias as created for provisioned concurrency case

Reply:
The combination of provisionedConcurrency and SnapStart is not supported by AWS and will give an error : `The operation cannot be performed at this time. Provisioned Concurrency is not supported for SnapStart enabled functions`
I have also mentioned this in [[docs/providers/aws/guide/functions.md](url)](https://github.com/serverless/serverless/blob/f914927d1b06d269ade2ce5ad7a0011f0b7c6e25/docs/providers/aws/guide/functions.md).

Reply:
Ok, I will add something similar like if you use both handler and image. It's indeed better to handle this up front then instead of waiting on a validation error from AWS side.

Reply:
Validation is added together with a test to validate this behaviour